src/Umbraco.Core/Persistence/Repositories/Implement/LanguageRepository.cs

```diff
@@ -262,17 +270,20 @@ public string GetDefaultIsoCode()
         private ILanguage GetDefault()
         {
             // get all cached, non-cloned
-            var all = TypedCachePolicy.GetAllCached(PerformGetAll);
+            var languages = TypedCachePolicy.GetAllCached(PerformGetAll).ToList();
+            var language = languages.FirstOrDefault(x => x.IsDefaultVariantLanguage);
+            if (language != null) return language;
+
+            // this is an anomaly, the service/repo should ensure it cannot happen
+            Logger.Warn<LanguageRepository>("There is no default language. Fix this anomaly by editing the language table in database and setting one language as the default language.");
+
+            // still, don't kill the site, and return "something"
 
             ILanguage first = null;
-            foreach (var language in all)
+            foreach (var l in languages)
             {
-                // if one language is default, return
-                if (language.IsDefaultVariantLanguage)
-                    return language;
-                // keep track of language with lowest id
-                if (first == null || language.Id < first.Id)
-                    first = language;
+                if (first == null || l.Id < first.Id)
```

Comment:
I know this comparison was there before but will this always work if the `languages` list isn't specifically sorted by id?